feature/experimental/p4rt/ate_tests/base_p4rt/base_p4rt_test.go

```diff
@@ -0,0 +1,413 @@
+			}
+		}
+	}
+
+	// Load p4info file.
+	p4Info, err := utils.P4InfoLoad(p4InfoFile)
+	if err != nil {
+		return errors.New("Errors seen when loading p4info file.")
+	}
+	setp4Info, _ := json.Marshal(p4Info)
+
+	// Send SetForwardingPipelineConfig for p4rt leader client.
+	fmt.Println("Sending SetForwardingPipelineConfig for both clients")
+	deviceid_list := []uint64{deviceId1, deviceId2}
+	for index, client := range clients {
+		if err := client.SetForwardingPipelineConfig(&p4_v1.SetForwardingPipelineConfigRequest{
+			DeviceId:   deviceid_list[index],
+			ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionId},
+			Action:     p4_v1.SetForwardingPipelineConfigRequest_VERIFY_AND_COMMIT,
+			Config: &p4_v1.ForwardingPipelineConfig{
+				P4Info: &p4Info,
+				Cookie: &p4_v1.ForwardingPipelineConfig_Cookie{
+					Cookie: 159,
+				},
+			},
+		}); err != nil {
+			return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+		}
+	}
+
+	// Receive GetForwardingPipelineConfig
+	for index, client := range clients {
+		resp, err := client.GetForwardingPipelineConfig(&p4_v1.GetForwardingPipelineConfigRequest{
+			DeviceId:     deviceid_list[index],
+			ResponseType: p4_v1.GetForwardingPipelineConfigRequest_P4INFO_AND_COOKIE,
+		})
+		if err != nil {
+			return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+		}
+		// Compare P4Info from GetForwardingPipelineConfig and SetForwardingPipelineConfig
+		getp4Info, _ := json.Marshal(resp.Config.P4Info)
+		if cmp.Equal(string(setp4Info), string(getp4Info)) {
+			fmt.Println("P4info matches fine for client", index)
+		} else {
+			err := fmt.Errorf("P4info does not match for client %d", index)
+			fmt.Println(err.Error())
+		}
+	}
+	return nil
+}
+
+// Function to compare and check if the expected table is present in RPC ReadResponse
+func verifyReadReceiveMatch(expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
+
+	matches := 0
+	for _, table := range received_entry.Entities {
+		if cmp.Equal(table.Entity, expected_update[0].Entity.Entity) {
+			fmt.Println("Table match succesful")
+			matches += 1
+			break
+		}
+	}
+	if matches == 0 {
+		return errors.New("match unsuccesful")
+	}
+	return nil
+}
+
+// TestP4rtConnect connects to the P4Runtime server over grpc
+// It then calls setupP4RTClient which sets the arbitration request and sends SetForwardingPipelineConfig with P4Info
+func TestP4rtConnect(t *testing.T) {
+
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+	ate := ondatra.ATE(t, "ate")
+
+	// configure DUT with P4RT node-id and ids on different FAPs
+	configureDUT(t, dut, []attrs.Attributes{dutPort1, dutPort2}, []string{*comp1name, *comp2name}, []uint64{deviceId1, deviceId2}, []uint32{portId1, portId2})
+
+	top := configureATE(t, ate)
+
+	// Setup two different clients for different FAPs
+	client1 := p4rt_client.P4RTClient{}
+	if err := client1.P4rtClientSet(dut.RawAPIs().P4RT(t)); err != nil {
```

Comment:
Noticing errors when I tried to run the test can we replace :
if err := client1.P4rtClientSet(dut.RawAPIs().P4RT(t)); err != nil { 
to 
if err := client1.P4rtClientSet(dut.RawAPIs().P4RT().Default(t)); err != nil { 
 which is currently being used by tests which are merged in master

Reply:
@rsindhujha can you fix this?

Reply:
Working on all fixes now.